imap_processing/hit/l2/hit_l2.py

```diff
@@ -46,64 +47,104 @@ def hit_l2(dependency_sci: xr.Dataset, dependencies_anc: list) -> list[xr.Datase
-        )
-        logger.info("HIT L2 macropixel intensity dataset created")
-
-    # Update attributes and dimensions
-    for logical_source, dataset in l2_datasets.items():
-        dataset.attrs = attr_mgr.get_global_attributes(logical_source)
-
-        # TODO: Add CDF attributes to yaml once they're defined for L2 science data
-        #  consider moving attribute handling to hit_utils.py
-        # Assign attributes and dimensions to each data array in the Dataset
-        for field in dataset.data_vars.keys():
-            try:
-                # Create a dict of dimensions using the DEPEND_I keys in the
-                # attributes
-                dims = {
-                    key: value
-                    for key, value in attr_mgr.get_variable_attributes(field).items()
-                    if "DEPEND" in key
-                }
-                dataset[field].attrs = attr_mgr.get_variable_attributes(field)
-                dataset[field].assign_coords(dims)
-            except KeyError:
-                # TODO: consider raising an error after L2 attributes are defined.
-                #  Until then, continue with processing and log warning
-                logger.warning(f"Field {field} not found in attribute manager.")
-
-        # Skip schema check for epoch to prevent attr_mgr from adding the
-        # DEPEND_0 attribute which isn't required for epoch
-        dataset.epoch.attrs = attr_mgr.get_variable_attributes(
-            "epoch", check_schema=False
-        )
+        l2_dataset = process_macropixel_intensity(dependency_sci, dependencies_anc)
+        logical_source = "imap_hit_l2_macropixel-intensity"
+
+    # Add attributes to dataset
+    if l2_dataset is not None and logical_source is not None:
+        l2_dataset = add_cdf_attributes(l2_dataset, logical_source, attr_mgr)
 
         logger.info(f"HIT L2 dataset created for {logical_source}")
 
-    return list(l2_datasets.values())
+    return [l2_dataset]
+
+
+def add_cdf_attributes(
+    dataset: xr.Dataset, logical_source: str, attr_mgr: ImapCdfAttributes
+) -> xr.Dataset:
+    """
+    Update attributes to the given dataset.
+
+    This function adds attributes to the dataset variables and dimensions.
+    It also adds dimension labels to the dataset as coordinates.
+
+    Parameters
+    ----------
+    dataset : xr.Dataset
+        The dataset to update.
+    logical_source : str
+        The logical source of the dataset.
+    attr_mgr : AttributeManager
+        The attribute manager to retrieve attributes.
+
+    Returns
+    -------
+    xr.Dataset
+        The updated dataset with attributes and dimension labels.
+    """
+    # Update global attributes
+    dataset.attrs = attr_mgr.get_global_attributes(logical_source)
+
+    # Assign attributes to each data variable in the Dataset
+    for var in dataset.data_vars.keys():
+        try:
+            if (
+                "macropixel" in logical_source
```

Comment:
It'd be helpful to have a comment here to make it more clear what this if check is looking for. It's not clear from the conditions what kind of variable would pass this check